apps/ima/fields/index.js

```diff
@@ -2,9 +2,243 @@
+  name: {
+    labelClassName: 'visuallyhidden',
+    validate: ['required'],
+    className: ['govuk-input', 'govuk-!-width-two-thirds']
+  },
+  'current-email': {
+    mixin: 'radio-group',
+    options: ['yes', 'no'],
+    validate: 'required',
+    legend: {
+      className: 'visuallyhidden'
+    }
+  },
+  'is-your-email': {
+    isPageHeading: true
+  },
+  'phone-number': {
+    isPageHeading: true,
+    mixin: 'radio-group',
+    validate: ['required'],
+    options: [{
+      value: 'yes',
+      toggle: 'phone-number-details',
+      child: 'input-text'
+    }, {
+      value: 'no'
+    }]
+  },
+  'phone-number-details': {
+    validate: ['required', 'internationalPhoneNumber', {type: 'maxlength', arguments: [200]}],
+    className: ['govuk-input', 'govuk-!-width-two-thirds'],
+    dependent: {
+      field: 'phone-number',
+      value: 'yes'
+    }
+  },
+  'email-address': {
+    isPageHeading: true,
+    mixin: 'radio-group',
+    validate: ['required'],
+    options: [{
+      value: 'yes',
+      toggle: 'email-address-details',
+      child: 'input-text'
+    }, {
+      value: 'no'
+    }]
+  },
+  'email-address-details': {
+    validate: ['required', 'email'],
+    className: ['govuk-input', 'govuk-!-width-two-thirds'],
+    dependent: {
+      field: 'email-address',
+      value: 'yes'
+    }
+  },
+  'has-address': {
+    isPageHeading: true,
+    mixin: 'radio-group',
+    validate: ['required'],
+    options: [{
+      value: 'yes'
+    },
+    {
+      value: 'no',
+      toggle: 'address-details-fieldset',
+      child: 'partials/address-details'
+    }]
+  },
+  'house-number': {
+    mixin: 'input-text',
+    validate: ['required', 'notUrl'],
+    includeInSummary: false,
+    validationLink: {
+      field: 'has-address',
+      value: 'no'
+    }
+  },
+  street: {
+    mixin: 'input-text',
+    validate: ['notUrl'],
+    includeInSummary: false,
+    validationLink: {
+      field: 'has-address',
+      value: 'no'
+    }
+  },
+  townOrCity: {
+    mixin: 'input-text',
+    validate: ['required', 'notUrl',
+      { type: 'regex', arguments: /^([^0-9]*)$/ },
+      { type: 'maxlength', arguments: 200 }
+    ],
+    className: ['govuk-input', 'govuk-!-width-two-thirds'],
+    validationLink: {
+      field: 'has-address',
+      value: 'no'
+    }
+  },
+  county: {
+    mixin: 'input-text',
+    validate: ['notUrl',
+      { type: 'regex', arguments: /^([^0-9]*)$/ },
+      { type: 'maxlength', arguments: 200 }
+    ],
+    includeInSummary: false,
+    className: ['govuk-input', 'govuk-!-width-two-thirds'],
+    validationLink: {
+      field: 'has-address',
+      value: 'no'
+    }
+  },
+  postcode: {
+    mixin: 'input-text',
+    validate: ['required', 'postcode', { type: 'maxlength', arguments: [200] }],
+    formatter: ['ukPostcode'],
+    includeInSummary: false,
+    className: ['govuk-input', 'govuk-input--width-10'],
+    validationLink: {
+      field: 'has-address',
+      value: 'no'
+    }
+  },
```

Comment:
Same validation issue as others. Also need to add for the other mandatory fields that are on the same page as the 'postcode' field.